compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1583,55 +1583,66 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
     }
 
     fn suggest_remove_await(&self, obligation: &PredicateObligation<'tcx>, err: &mut Diagnostic) {
-        let span = obligation.cause.span;
-
-        if let ObligationCauseCode::AwaitableExpr(hir_id) = obligation.cause.code().peel_derives() {
-            let hir = self.tcx.hir();
-            if let Some(hir::Node::Expr(expr)) = hir_id.and_then(|hir_id| hir.find(hir_id)) {
-                // FIXME: use `obligation.predicate.kind()...trait_ref.self_ty()` to see if we have `()`
-                // and if not maybe suggest doing something else? If we kept the expression around we
-                // could also check if it is an fn call (very likely) and suggest changing *that*, if
-                // it is from the local crate.
+        let hir = self.tcx.hir();
+        if let ObligationCauseCode::AwaitableExpr(Some(hir_id)) = obligation.cause.code().peel_derives()
+            && let hir::Node::Expr(expr) = hir.get(*hir_id)
+        {
+            // FIXME: use `obligation.predicate.kind()...trait_ref.self_ty()` to see if we have `()`
+            // and if not maybe suggest doing something else? If we kept the expression around we
+            // could also check if it is an fn call (very likely) and suggest changing *that*, if
+            // it is from the local crate.
+
+            if let hir::Node::Expr(parent_expr) = hir.get_parent(*hir_id)
+                // Peel off the DesugaringKind from the span
+                && let Some(desugar_parent_span) = parent_expr.span.parent_callsite()
+            {
+                let removal_span = self.tcx
+                    .sess
+                    .source_map()
+                    .span_extend_while(expr.span, char::is_whitespace)
+                    .unwrap_or(expr.span)
+                    .shrink_to_hi()
+                    .to(desugar_parent_span);
```

Comment:
> Do we need to care about whitespace?

Trimming the whitespace makes the suggestion a lot easier to read in the cases of suggesting remove `.await` in:

```
very_very_long_expr
    .await
```

> Should we find expr.span's ancestor within desugar_parent_span, in case of $expr.await?

Yeah probably. I'll create a test for that.